### PR TITLE
Default auth handling fixes

### PR DESCRIPTION
**Pull request by kunagpal**

### CODE REVIEW

File: lib/normalizers/v1.js [hunk 1]
```diff
@@ -59,8 +59,7 @@ _.assign(Builders.prototype, {
         if (util.notLegacy(entityV1, 'auth')) {
             return util.sanitizeAuthArray(entityV1, options);
         }
-        if (entityV1.currentHelper === null) { return this.options.noDefaults ? undefined : { type: 'noauth' }; }
-        if (entityV1.currentHelper === 'normal') { return { type: 'noauth' }; }
+        if ((entityV1.currentHelper === null) || (entityV1.currentHelper === 'noauth')) { return null; }
```
Comment (kamalaknn) [inline lib/normalizers/v1.js]: (entityV1.currentHelper === null) || (entityV1.currentHelper === 'normal')

File: lib/normalizers/v1.js [hunk 2]
```diff
@@ -242,7 +241,13 @@ _.assign(Builders.prototype, {
 
         auth = self.auth(requestV1);
 
-        if (auth) { requestV1.auth = auth; }
+        if (auth) {
+            requestV1.auth = auth;
+
+            if (_.has(requestV1, 'helperAttributes') && !requestV1.currentHelper) {
```
Comment (kamalaknn) [inline lib/normalizers/v1.js]: Can you make this in a separate PR? Is this blocking this PR?

File: lib/util.js [hunk 3]
```diff
@@ -202,7 +202,7 @@ util = {
             case 'event':
                 return !(entityV1.tests || entityV1.preRequestScript);
             case 'auth':
-                return _.has(entityV1, 'auth') && !(_.has(entityV1, 'currentHelper') && entityV1.helperAttributes);
+                return _.has(entityV1, 'auth') && !(_.has(entityV1, 'currentHelper') || entityV1.helperAttributes);
```
Comment (kamalaknn) [inline lib/util.js]: Why this change?

File: lib/normalizers/v1.js (hunk 2 above)
Comment (kunagpal) [reply lib/normalizers/v1.js]: Yes.

File: lib/util.js (hunk 3 above)
Comment (kunagpal) [reply lib/util.js]: This was a bug in the legacy mode detection.

Comment (kunagpal) [reply lib/util.js]: The change above is related to this.

Comment (kamalaknn) [reply lib/util.js]: Got it

File: test/unit/v1.0.0/normalize-v1.test.js [hunk 4]
```diff
@@ -96,6 +98,221 @@ describe('v1.0.0 normalization', function () {
...
+                    });
+                });
+
+                describe('currentHelper: normal', function () {
+                    describe('noDefaults: false', function () {
+                        it('should handle auth (null) and currentHelper(normal) correctly', function (done) {
+                            transformer.normalizeSingle({
+                                id: '27c8c9ac-dd90-4234-b83a-b199d3a0e945',
+                                auth: null,
+                                currentHelper: null
```
Comment (kamalaknn) [inline test/unit/v1.0.0/normalize-v1.test.js]: description says `currentHelper(normal)` but test is on `currentHelper(null)`?
